### ISSUE TITLE
Error when calling "hcloud_server": TypeError: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by papanito** [ansible-collections/hetzner.hcloud#73]
##### SUMMARY

While I run `hcloud_server` I get an `TypeError: 'NoneType' object is not callable` even so everything seems set.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`hetzner.hcloud`

Versions of collections:
- `community.hrobot:1.1.1`
- `community.general:2.4.0`

##### ANSIBLE VERSION

```paste below
ansible 2.10.7
  config file = /home/aedu/ansible.cfg
  configured module search path = ['/home/aedu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.9.2 (default, Feb 20 2021, 18:40:11) [GCC 10.2.0]
```

##### CONFIGURATION

```paste below
DEFAULT_HOST_LIST(/home/aedu/ansible.cfg) = ['/home/aedu/inventory.yml']
DEFAULT_REMOTE_USER(/home/aedu/ansible.cfg) = ansible
DEFAULT_ROLES_PATH(/home/aedu/ansible.cfg) = ['/home/aedu/roles']
DEFAULT_TIMEOUT(/home/aedu/ansible.cfg) = 10
DEFAULT_VAULT_PASSWORD_FILE(/home/aedu/ansible.cfg) = /home/aedu/.ssh/mykey
GALAXY_SERVER_LIST(/home/aedu/ansible.cfg) = ['release_galaxy']
RETRY_FILES_SAVE_PATH(/home/aedu/ansible.cfg) = /home/aedu/.ansible-retry
```

##### OS / ENVIRONMENT

`Linux 5.11.10-1-MANJARO x86_64 GNU/Linux`

##### STEPS TO REPRODUCE

Run `ansible-playbook demo.yml --ask-become-pass --check -vvvv`

```yaml
- hosts: localhost
  gather_facts: no

  pre_tasks:
  - name: Create a basic server
    hcloud_server:
      api_token: "{{ hcloud_token }}"
      name: demo
      server_type: cx11
      image: debian-10
      state: present
      ssh_keys: 
        - ansible-user
        - aedu-clawfinger
    delegate_to: localhost
    register: result
```

##### EXPECTED RESULTS

Server instance is created in Hetzner Cloud

##### ACTUAL RESULTS

```paste below
> ansible-playbook demo.yml  --ask-become-pass  -vvvv --check
ansible-playbook 2.10.7
  config file = /home/aedu/Workspaces/wyssmann.com/infrastructure/ansible.cfg
  configured module search path = ['/home/aedu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 3.9.2 (default, Feb 20 2021, 18:40:11) [GCC 10.2.0]
Using /home/aedu/Workspaces/wyssmann.com/infrastructure/ansible.cfg as config file
BECOME password: 
setting up inventory plugins
host_list declined parsing /home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.yml as it did not pass its verify_file() method
script declined parsing /home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.yml as it did not pass its verify_file() method
Parsed /home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.yml inventory source with yaml plugin
setting up inventory plugins
host_list declined parsing /home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.hcloud.yml as it did not pass its verify_file() method
script declined parsing /home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.hcloud.yml as it did not pass its verify_file() method
redirecting (type: inventory) ansible.builtin.hcloud to hetzner.hcloud.hcloud
Loading collection hetzner.hcloud from /home/aedu/.ansible/collections/ansible_collections/hetzner/hcloud
/home/aedu/.local/lib/python3.9/site-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.25.11) or chardet (4.0.0) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
Parsed /home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.hcloud.yml inventory source with auto plugin
redirecting (type: modules) ansible.builtin.hcloud_server to hetzner.hcloud.hcloud_server
Loading callback plugin default of type stdout, v2.0 from /usr/lib/python3.9/site-packages/ansible/plugins/callback/default.py
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.

PLAYBOOK: demo.yml **********************************************************************************************************************************************************************************************
Positional arguments: demo.yml
verbosity: 4
remote_user: ansible
connection: smart
timeout: 10
become_method: sudo
become_ask_pass: True
tags: ('all',)
check: True
diff: True
inventory: ('/home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.yml', '/home/aedu/Workspaces/wyssmann.com/infrastructure/inventory.hcloud.yml')
forks: 5
1 plays in demo.yml

PLAY [localhost] ************************************************************************************************************************************************************************************************

TASK [Create a basic server] ************************************************************************************************************************************************************************************
task path: /home/aedu/Workspaces/wyssmann.com/infrastructure/demo.yml:5
Trying secret FileVaultSecret(filename='/home/aedu/.ssh/wyssmann') for vault_id=default
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: aedu
<localhost> EXEC /bin/sh -c 'echo ~aedu && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/aedu/.ansible/tmp `"&& mkdir "` echo /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925 `" && echo ansible-tmp-1617739488.1029868-1196900-212474867712925="` echo /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925 `" ) && sleep 0'
redirecting (type: modules) ansible.builtin.hcloud_server to hetzner.hcloud.hcloud_server
Using module file /home/aedu/.ansible/collections/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py
<localhost> PUT /home/aedu/.ansible/tmp/ansible-local-11967809agsx6h7/tmpy_ffby82 TO /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/ /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py && sleep 0'
<localhost> EXEC /bin/sh -c 'sudo -H -S  -p "[sudo via ansible, key=cihudlxmuqytobocpmmbmtdwjbpfigpx] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-cihudlxmuqytobocpmmbmtdwjbpfigpx ; /usr/bin/python /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py'"'"' && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py", line 102, in <module>
    _ansiballz_main()
  File "/home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.hetzner.hcloud.plugins.modules.hcloud_server', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.9/runpy.py", line 210, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py", line 616, in <module>
  File "/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py", line 597, in main
  File "/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py", line 540, in present_server
  File "/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py", line 333, in _create_server
  File "/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py", line 334, in <listcomp>
TypeError: 'NoneType' object is not callable
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/aedu/.ansible/tmp/ansible-tmp-1617739488.1029868-1196900-212474867712925/AnsiballZ_hcloud_server.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.hetzner.hcloud.plugins.modules.hcloud_server', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib/python3.9/runpy.py\", line 210, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 616, in <module>\n  File \"/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 597, in main\n  File \"/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 540, in present_server\n  File \"/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 333, in _create_server\n  File \"/tmp/ansible_hcloud_server_payload_r5b303kq/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 334, in <listcomp>\nTypeError: 'NoneType' object is not callable\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP ******************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

**Comment by LKaemmerling** [comment]
Hey @papanito,

could you please try it with the latest version (1.4.1)?

**Comment by papanito** [comment]
Maybe I miss something:

```
ERROR! Failed to find collection hetzner.hcloud:1.4.1
```
My `requriements.yml`

```
....
collections:
  - name: hetzner.hcloud
    version: 1.4.1
  - name: community.hrobot
  - name: community.general
```

**Comment by LKaemmerling** [comment]
Yeah give it a few more minutes :) https://dashboard.zuul.ansible.com/t/ansible/stream/9d273f508dcd411ebe1a9292520cd457?logfile=console.log 

The Job is still processing

**Comment by papanito** [comment]
Yeah just realized it. Thanks @LKaemmerling

**Comment by papanito** [comment]
Still the same:

```
Traceback (most recent call last):\n  File \"/home/aedu/.ansible/tmp/ansible-tmp-1617777252.9323785-1236366-237186456248268/AnsiballZ_hcloud_server.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/aedu/.ansible/tmp/ansible-tmp-1617777252.9323785-1236366-237186456248268/AnsiballZ_hcloud_server.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/aedu/.ansible/tmp/ansible-tmp-1617777252.9323785-1236366-237186456248268/AnsiballZ_hcloud_server.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.hetzner.hcloud.plugins.modules.hcloud_server', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib/python3.9/runpy.py\", line 210, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_hcloud_server_payload_ufhof_5p/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 645, in <module>\n  File \"/tmp/ansible_hcloud_server_payload_ufhof_5p/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 626, in main\n  File \"/tmp/ansible_hcloud_server_payload_ufhof_5p/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 568, in present_server\n  File \"/tmp/ansible_hcloud_server_payload_ufhof_5p/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 334, in _create_server\n  File \"/tmp/ansible_hcloud_server_payload_ufhof_5p/ansible_hcloud_server_payload.zip/ansible_collections/hetzner/hcloud/plugins/modules/hcloud_server.py\", line 335, in <listcomp>\nTypeError: 'NoneType' object is not callable\n"
```

**Comment by LKaemmerling** [comment]
@papanito do you have the latest hcloud-python version installed? https://github.com/hetznercloud/hcloud-python/releases/tag/v1.12.0 (or v1.11.0)

**Comment by papanito** [comment]
Nope I have 
```bash
pacman -Q  hcloud-python
python-hcloud 1.10.0-2
```

**Comment by LKaemmerling** [comment]
Then this is the issue. The ansible integration v1.3.0+ require the hcloud-python 1.11.0+ (v1.4.0 requires hcloud-python 1.12.0+)

**Comment by papanito** [comment]
I removed the os-package and installed the latest package via `pip install hcloud==1.12.0`. Now the error is gone.
@LKaemmerling thanks a lot for your quick help!

**Comment by papanito** [comment]
Two more things:

- The dependency is not fully clear when looking at https://docs.ansible.com/ansible/latest/collections/hetzner/hcloud/hcloud_inventory.html#ansible-collections-hetzner-hcloud-hcloud-inventory
- I have seen `"firewalls": null,` in the output of the playbook under `module_args` does this mean one could also set a firewall? not have seen anything in the docu yet ;-)

**Comment by LKaemmerling** [comment]
@papanito yep since v1.3.0 we support our Firewalls: https://www.hetzner.com/news/03-21-firewall-beta/


The document you linked is for the collection in v1.2.1 (https://docs.ansible.com/ansible/latest/collections/hetzner/hcloud/hcloud_inventory.html#ansible-collections-hetzner-hcloud-hcloud-inventory) not for the latest 1.4.1.

As the problem is solved, I will close the issue :) Feel free to reopen if you have further questions

**Comment by papanito** [comment]
Is there a documentation available for 1.4.1?

**Comment by LKaemmerling** [comment]
Hey @papanito,

you can use `ansible-doc hetzner.hcloud.<module>`
